engine/src/java/com/runrev/android/billing/google/GoogleBillingProvider.java

```diff
@@ -496,14 +496,25 @@ public void onQueryInventoryFinished(IabResult result, Inventory inventory)
 
             Log.d(TAG, "Initial inventory query finished; enabling main UI.");
 
+			// PM-2015-02-05: [[ Bug 14402 ]] Display a msg in case there are no previous purchases to restore
+			boolean t_did_restore;
+			t_did_restore = false;
+
             List<Purchase> purchaseList = inventory.getallpurchases();
+
             for (Purchase p : purchaseList)
             {
                 addPurchaseToLocalInventory(p);
 				ownedItems.add(p.getSku());
 				// onPurchaseStateChanged to be called with state = 5 (restored)
 				mPurchaseObserver.onPurchaseStateChanged(p.getSku(), 5);
+				t_did_restore = true;
             }
+
+			if(!t_did_restore)
+			{
```

Comment:
I don't think this is the right thing to do... We are showing an alert from a callback or API directly - this cannot be localised. In general the engine should never popup a blocking dialog - it should send a message so that the LiveCode script can handle the case and act accordingly.